extensions/amp-dailymotion/dailymotion-api.js

```diff
@@ -0,0 +1,95 @@
+import {addParamsToUrl} from '../../src/url';
+import {dict} from '#core/types/object';
+import {isAutoplaySupported} from '#core/dom/video';
+
+/**
+ *
+ * @param {!Window} win
+ * @param {string} videoId
+ * @param {boolean} autoplay
+ * @param {boolean} endscreenEnable
+ * @param {boolean} info
+ * @param {boolean} mute
+ * @param {boolean} sharingEnable
+ * @param {string} start
+ * @param {string} uiHighlight
+ * @param {boolean} uiLogo
+ * @param {!JsonObject} implicitParams
+ * @return {string}
+ */
+export function getDailymotionIframeSrc(
+  win,
+  videoId,
+  autoplay,
+  endscreenEnable,
+  info,
+  mute,
+  sharingEnable,
+  start,
+  uiHighlight,
+  uiLogo,
+  implicitParams
+) {
+  return addParamsToUrl(
+    `https://www.dailymotion.com/embed/video/${encodeURIComponent(
+      videoId
+    )}?api=1&html=1&app=amp`,
+    Object.assign(
+      dict({
+        'endscreen-enable': endscreenEnable,
+        'info': info,
+        // In order to support autoplay the video needs to be muted on load so we
+        // dont receive an unmute event which prevents the video from autoplay.
+        'mute': mute || (autoplay && isAutoplaySupported(win)) ? 1 : undefined,
+        'sharing-enable': sharingEnable,
+        'start': start,
+        'ui-highlight': uiHighlight,
+        'ui-logo': uiLogo,
+      }),
+      implicitParams
+    )
+  );
+}
+
+/**
+ * @param {string} command
+ * @param {?Object|string=} params
```

Comment:
I noticed all applications of this helper in 0.1 and 1.0 only ever pass one boolean, if any. Is the array really necessary? Can we update calls to only pass the one optional param?
```suggestion
 * @param {?boolean} param
```